src/commands/synthetics/run-tests-lib.ts

```diff
@@ -191,53 +176,86 @@ const getTestListBySearchQuery = async (
     suite: `Query: ${testSearchQuery}`,
   }))
 }
-
-export const getTestsList = async (
+export const getTriggerConfigs = async (
   api: APIHelper,
   config: RunTestsCommandConfig,
   reporter: MainReporter,
-  suites: Suite[] = []
-) => {
-  // If "testSearchQuery" is provided, always default to running it.
-  if (config.testSearchQuery) {
-    const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery)
-
-    if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
-      reporter.error(
-        `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
-      )
+  suites?: Suite[]
+): Promise<TriggerConfig[]> => {
+  const defaultTestConfigOverrides = config.global
+  // TODO: Clean up locations as part of SYNTH-12989
+  const testConfigOverridesFromEnv = config.locations?.length ? {locations: config.locations} : {}
+  const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter) : []
+  const testsFromTestConfigs = await getTestConfigs(config, reporter, suites)
+
+  // Grab the list of publicIds of tests to trigger from config file/env variables/CLI params, search query or test config files
+  const testIdsFromCli = config.publicIds
+  const testIdsFromSearchQuery = testsFromSearchQuery.map(({id}) => id)
+  const testIdsFromTestConfigs = testsFromTestConfigs.map(({id}) => id)
+
+  const testIdsToTrigger =
+    testIdsFromCli.length !== 0
+      ? testIdsFromCli
+      : testIdsFromSearchQuery.length !== 0
+      ? testIdsFromSearchQuery
+      : testIdsFromTestConfigs.length !== 0
+      ? testIdsFromTestConfigs
+      : []
```

Comment:
After giving it some thought, we might have been going in the wrong direction with this last change.
If the search query is defined but returns no tests, should it:
- continue with the test configs?
- return an empty list to throw the 'no tests to run' error later?
It might more intuitive for it to be the last case, so we can really use `undefined` to mean that the list is not defined, and `[]` to mean that the list is defined but contain no tests, which will lead to an error.

In the latter case, we might want to add these modifications:
```suggestion
    testIdsFromCli ??  testIdsFromSearchQuery ?? testIdsFromTestConfigs ?? []
```

```diff
- const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter) : []
+ const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter)
```

```diff
- const testFromSearchQuery = testsFromSearchQuery.find((test) => test.id === id)
+ const testFromSearchQuery = testsFromSearchQuery?.find((test) => test.id === id)
```

Reply:
If you give a wrong `--public-id` in the CLI, so no such test was found, we return `no tests to run` so it does make sense that if you give a `--search` that returns nothing for it to also give `no tests to run`, which would be different than not having a `--search` defined at all in the first place, yes 🤔

Reply:
We got together and found a nice way to do it 👍